### PR TITLE
chore: make vcr tests pretty-print JSON

### LINKED ISSUE #127: Investigate formatting multiline values in cassette yaml files
In #118 we added [VCR] tests and the generated `yaml` files are difficult to read.

```yaml
      string: "{\n  \"candidates\": [\n    {\n      \"content\": {\n        \"parts\":
        [\n          {\n            \"text\": \"Hello! \U0001F44B  How can I help
        you today? \U0001F60A \\n\"\n          }\n        ],\n        \"role\": \"model\"\n
        \     },\n      \"finishReason\": \"STOP\",\n      \"index\": 0,\n      \"safetyRatings\":
        [\n        {\n          \"category\": \"HARM_CATEGORY_SEXUALLY_EXPLICIT\",\n
        \         \"probability\": \"NEGLIGIBLE\"\n        },\n        {\n          \"category\":
        \"HARM_CATEGORY_HATE_SPEECH\",\n          \"probability\": \"NEGLIGIBLE\"\n
        \       },\n        {\n          \"category\": \"HARM_CATEGORY_HARASSMENT\",\n
        \         \"probability\": \"NEGLIGIBLE\"\n        },\n        {\n          \"category\":
        \"HARM_CATEGORY_DANGEROUS_CONTENT\",\n          \"probability\": \"NEGLIGIBLE\"\n
        \       }\n      ]\n    }\n  ],\n  \"usageMetadata\": {\n    \"promptTokenCount\":
        8,\n    \"candidatesTokenCount\": 12,\n    \"totalTokenCount\": 20\n  }\n}\n"
```

Ideally, the output would be a YAML literal block scalar for readability

```yaml
interactions:
  - request:
      body: |-
        {
          "system_instruction": {
            "parts": [
              {
                "text": "You are a helpful assistant."
              }
            ]
          },
          "contents": [
            {
              "role": "user",
              "parts": [
                {
                  "text": "Hello"
                }
              ]
            }
          ]
        }
  - response:
      body:
        string: |- 
          {
            "candidates": [
              {
                "content": {
                  "parts": [
                    {
                      "text": "Hello! 👋  How can I help you today? 😊 \n"
                    }
                  ],
                  "role": "model"
                },
                "finishReason": "STOP",
                "index": 0,
                "safetyRatings": [ ... ]
                ]
              }
            ],
            "usageMetadata": { ... }
          }
          
```

[VCR]: https://vcrpy.readthedocs.io/en/latest/

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This makes recordings of HTTP requests and responses more readable at the cost of a little more code.

cc @lamchau @anuraaga as I'm still learning python, so possibly my approach isn't ideal. Changes welcome!

Fixes #127

### CODE REVIEW

File: packages/exchange/tests/providers/conftest.py [hunk 1]
```diff
@@ -70,6 +73,70 @@ def default_google_env(monkeypatch):
         monkeypatch.setenv("GOOGLE_API_KEY", GOOGLE_API_KEY)
 
 
+# Use a literal string for long or multi-line strings
+class LiteralString(str):
+    pass
+
+
+def literal_presenter(dumper, data):
```
Comment (anuraaga) [inline packages/exchange/tests/providers/conftest.py]: Maybe a comment on what this does. IIUC it forces mutliline string

**Review by anuraaga — APPROVED** [review]
Nice improvement

File: packages/exchange/tests/providers/conftest.py (hunk 1 above)
Comment (codefromthecrypt) [reply packages/exchange/tests/providers/conftest.py]: thanks for the feedback, I changed the function name and put some comments on it. hope it is better!

**Review by michaelneale — APPROVED** [review]

**Comment by michaelneale** [comment]
looks tidy

**Comment by lamchau** [comment]
@codefromthecrypt this is so much cleaner! very slick approach

**Review by lamchau — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
thanks for the look, folks

**Pull request merged**